File: LAB2/OP/5/csharp lab5/csharp lab5/TPyramid.cs
using System;
namespace csharp_lab5
{
    internal class TPyramid : TFigure
    {
        private double a, p, h;
        public TPyramid(double a, double p)
        {
            this.a = a;
            this.p = p;
        }
        public override double S()
        {
            return (p * a) / 2;
        }
        
        public override double V()
        {
            return (p * a) / 2 * (1/3)  * h;
        }

        public double A { get => a; set => a = value;}
        public double P { get => p; set => p = value;}
        
       public override string ToString()
       {
           return "Pyramid: " +
                  $"apothem = {a} " +
                  $"perimeter = {p} " +
                  $"Square = {Math.Round(S(),3)}";
       }
    }
}

File: LAB2/OP/5/py/TPyramid.py
from TFigure import *


class TPyramid(TFigure):

    def S(self):
        return (self.p * self.a) / 2

    def V(self):
        return (self.p * self.a) / 2 / 2 * (1/3) * self.h

    def __init__(self, a, p):
        self.a = a
        self.p = p

    def __str__(self):
        return "Pyramid: " + f"apothem = {self.a} " + f"perimeter = {self.p} " + f"Square = {self.S()} "
